publication-commons/src/test/java/no/unit/nva/service/impl/RestPublicationServiceTest.java

```diff
@@ -0,0 +1,205 @@
+package no.unit.nva.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.Environment;
+import no.unit.nva.PublicationHandler;
+import no.unit.nva.model.Publication;
+import no.unit.nva.service.PublicationService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.UUID;
+
+import static no.unit.nva.service.impl.RestPublicationService.API_HOST_ENV;
+import static no.unit.nva.service.impl.RestPublicationService.API_SCHEME_ENV;
+import static no.unit.nva.service.impl.RestPublicationService.NOT_IMPLEMENTED;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RestPublicationServiceTest {
+
+    public static final String PUBLICATION_JSON = "src/test/resources/publication.json";
```

Comment:
If you use [this](https://github.com/BIBSYSDEV/aws-build-tools/blob/master/src/main/java/no/bibsys/aws/tools/IoUtils.javaclass) class you can read a reousrce directly without using OS dependent path separators and implementation dependent paths